client/reader/controller.js

```diff
@@ -318,3 +318,29 @@ export function readFollowingP2( context, next ) {
 	/* eslint-enable wpcalypso/jsx-classname-namespace */
 	next();
 }
+
+export async function blogDiscoveryByFeedId( context, next ) {
+	const { blog, feed_id } = context.params;
+
+	// If we have already had blog or we don't have feed_id, call `next()` immediately
+	if ( blog || ! feed_id ) {
+		next();
+		return;
+	}
+
+	// Query the site by feed_id and inject to the context params so that calypso can get correct site
+	// after redirecting the user to log-in page
+	context.queryClient
+		.fetchQuery(
+			[ '/read/feed/', feed_id ],
+			() => wpcom.req.get( `/read/feed/${ feed_id }` ).then( ( res ) => res.blog_ID ),
+			{ meta: { persist: false } }
+		)
+		.then( ( blog_id ) => {
+			context.params.blog = blog_id;
```

Comment:
Small detail, but I think `blog_id` is clearer (even though I see that we use `site` instead of `site_id` 🙈)

Reply:
Yup...there are legacy codes using `site`, `siteid`, `site_id`, `blog`, `blogid`, `blog_id` 🙈 But I'll change to `blog_id` here!

Reply:
I think it was originally `blog_id` and then when the v1 WordPress.com API launched, a decision was made to use `site_id` for the endpoints I think... so they are used interchangeably all over the place 🙈

Like @taipeicoder, I try and use `blog_id` consistently these days.